### DISCUSSION TITLE
[General] Sometines interacting with zksync returns an "invalid value for value.blockHash" Error

### DISCUSSION THREAD

**Discussion by cabreraction** [zkSync-Community-Hub/zksync-developers#136] [General] [answered]
### Environment

Testnet

### Provide a brief description of the functionality you're trying to implement.

Sometimes (not everytime) when using a smart contract (I've used different from differnt apps) to execute some transaction the transaction returns a "invalid value for value.blockHash". Is there a way around this? I haven't found anything else online 

### What is the specific issue or error you're encountering?

"invalid value for value.blockHash"

### Can you share the error messages or logs you're receiving, if any?

invalid value for value.blockHash (invalid hash (argument="value", value=null, code=INVALID_ARGUMENT, version=6.7.1)) (value={ "blockHash": null, "blockNumber": null, "contractAddress": null, "cumulativeGasUsed": "0x0", "effectiveGasPrice": "0x0", "from": "0xdbb0b721be26137f23a7ab8879c716ab663bf716", "gasUsed": "0x4c4b40", "l1BatchNumber": null, "l1BatchTxIndex": null, "l2ToL1Logs": [  ], "logs": [  ], "logsBloom": "0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000", "root": null, "status": null, "to": "0x1ed29c10e661e5721dfe162845f72548f090d8e7", "transactionHash": "0x2948d0c894b4108170baee1b630b7ee77d18970c80b99994ec4310834ceb28e4", "transactionIndex": "0x0", "type": "0x2" }, code=BAD_DATA, version=6.7.1)

### Have you made any recent changes to the contract before encountering this issue?

I dont own the contracts

### Are there any external libraries or contracts that your contract interacts with?

I dont own the contracts

### Can you provide the relevant portions of your contract code where the issue is occurring?

I dont own the contracts

### Have you tried to isolate the problem, and if so, what were the results?

I dont own the contracts

### What steps have you already taken to try to resolve the issue?

I dont own the contracts

**Comment by idea404** [comment]
Hi @cabreraction could you please provide the code for your smart contract that returns this error?

**Reply by MexicanAce** [reply, marked as answer]
Closing this stale discussion as there have been no responses in 2 weeks. If this issue re-occurs, please let us know!

**Comment by inesarmetessa** [comment]
This error typically occurs when the transaction receipt does not contain the expected blockHash value. It could be caused by issues with the node or network, transient network problems, or even a bug in the contract or transaction processing logic. Since you don't own the contracts and can't modify the code, your options are limited. You can try resending the transaction or waiting for the network to stabilize. If the issue persists, you may need to contact the developers or maintainers of the contracts or the blockchain infrastructure for assistance.

**Comment by dawn71v** [comment]
The error "invalid value for value.blockHash" typically indicates an issue with the block data returned by the blockchain node. It could be caused by various factors such as network instability or node misconfiguration. Try switching to a different node provider or updating your node software to see if the issue persists.

**Comment by loal0cksibus** [comment]
This error suggests that there's an issue with retrieving the block hash associated with the transaction. It could be due to network congestion, node synchronization issues, or intermittent errors. Try resubmitting the transaction or connecting to a different node to see if the problem persists. If the issue persists, consider reaching out to the developers of the contract or the platform for further assistance.
